src/modules/popUpDom.js

```diff
@@ -0,0 +1,45 @@
+import getData from './getdata.js';
+
+const popUpContainer = document.querySelector('.popUp-bg');
+const popUp = document.createElement('div');
+
+const displaySelectedMeal = (selectedMeal) => {
+  popUp.className = 'popUp';
+  popUp.innerHTML = `
+  <button class="closeBtn">X</button>
+  <img class="popUpImg" src="${selectedMeal.strCategoryThumb}" alt="meal image">
+  <h2 class="popUpTitle">${selectedMeal.strCategory}</h2>
+  <p class="popUpDiscription">${selectedMeal.strCategoryDescription}</p>
+  <h3 class="commentCount">Comments<span class="commentCounter">()</span></h3>
+  <div class="comment">
+      <p class="commentDate">03/11/2022</p>
+      <p class="commenterName">Khusniddin:</p>
+      <p class="commentContent">Lorem ipsum dolor sit amet.</p>
+  </div>
+  <h3 class="addComment">Add a comment</h3>
+      <form class="inputComment" action="#" method="post">
+          <input type="text" name="username" placeholder="Your name" required>
+          <textarea type="text" name="comment" placeholder="Your insights" required></textarea>
+          <button class="commentBtn" type="submit">Comment</button>
+      </form>`;
+
+  popUpContainer.appendChild(popUp);
+};
+
+const getSelectedMeal = async (selectedIdCat) => {
+  const selectedMeals = await getData();
+  if (selectedIdCat !== undefined) {
+    displaySelectedMeal(selectedMeals.categories.find((meal) => meal.idCategory === selectedIdCat));
+  }
```

Comment:
Can You explain why you use this method for selectedMeals fucntions.

Reply:
@soesandarwin2201 I am using this function to retrieve the id of the parent element of the comment button and I am comparing it with the id of the meal category so that I can show the right meal on the pop-up. If there is need for any improvements, please let me know

Reply:
@KIsmoilov  Thanks for your explanation , now i can use that method for my like Button function ✨